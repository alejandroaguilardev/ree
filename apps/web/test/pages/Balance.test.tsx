import React from 'react';
import { render } from '@testing-library/react';
import { describe, it, expect, vi } from 'vitest';
import BalancePage from '../../src/pages/Balance';

vi.mock('../../src/sections/Balance/hooks/useQueryBalance', () => ({
    useGetEnergyBalanceByDateRange: vi.fn().mockReturnValue({
        data: undefined,
        loading: false,
        error: undefined,
        refetch: vi.fn()
    }),
}));

describe('BalancePage Snapshot', () => {
    it('should match the snapshot', () => {


        const { asFragment } = render(<BalancePage />);
        expect(asFragment()).toMatchSnapshot();
    });
});
