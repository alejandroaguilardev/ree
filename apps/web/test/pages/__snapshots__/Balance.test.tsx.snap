// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BalancePage Snapshot > should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="bg-white rounded-lg   p-4 mb-6"
  >
    <div
      class="flex items-center gap-2 mb-3"
    >
      <p
        class="text-2xl font-semibold leading-snug text-black undefined"
      >
        Filtrar por rango de fechas
      </p>
    </div>
    <form
      role="form"
    >
      <div
        class="flex flex-row gap-4"
      >
        <div
          class="space-y-2 w-full"
        >
          <div
            class="w-full"
          >
            <label
              class="block text-sm font-medium text-gray-700 mb-1"
              for="startDate"
            >
              Fecha Inicial
            </label>
            <input
              class="w-full mt-1 block w-full px-3 py-2 border border-gray-300 rounded-md shadow-sm focus:outline-none focus:ring-indigo-500 focus:border-indigo-500 sm:text-sm"
              id="startDate"
              name="startDate"
              type="date"
              value="2025-04-26"
            />
          </div>
        </div>
        <div
          class="space-y-2 w-full"
        >
          <div
            class="w-full"
          >
            <label
              class="block text-sm font-medium text-gray-700 mb-1"
              for="endDate"
            >
              Fecha Final
            </label>
            <input
              class="w-full mt-1 block w-full px-3 py-2 border border-gray-300 rounded-md shadow-sm focus:outline-none focus:ring-indigo-500 focus:border-indigo-500 sm:text-sm"
              id="endDate"
              name="endDate"
              type="date"
              value="2025-04-26"
            />
          </div>
        </div>
        <div
          class="flex items-end w-full"
        >
          <button
            class="inline-flex items-center justify-center rounded-md text-sm font-medium transition-colors focus:outline-none focus:ring-2 focus:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-blue-500 text-white hover:bg-blue-400 h-10 px-4 py-2 w-full"
            type="submit"
          >
            Aplicar Filtros
          </button>
        </div>
      </div>
    </form>
  </div>
  <div
    class="flex flex-col items-center justify-center p-6 space-y-4"
  >
    <h2
      class="text-xl text-gray-600 font-semibold"
    >
      No se encontraron datos
    </h2>
    <p
      class="text-gray-500 text-center"
    >
      Parece que no hay información disponible para el rango seleccionado. Por favor, ajusta las fechas o intenta nuevamente más tarde.
    </p>
    <button
      class="px-6 py-2 bg-blue-500 text-white rounded-lg hover:bg-blue-600 transition duration-300"
    >
      Reintentar
    </button>
  </div>
</DocumentFragment>
`;
